network/src/transport.rs

```diff
@@ -397,3 +420,151 @@ pub fn build_tcp_transport(
         .with_timeout(TRANSPORT_TIMEOUT)
         .boxed()
 }
+
+#[cfg(test)]
```

Comment:
feel mixed about these tests, I don't know if we benefit from adding them to the code base, it is a lot to read and potentially modify... Ideally we cover these from a unit test perspective in the HandshakeMsg. Then we also handle it from our various integration and smoke tests. Thoughts? 150 lines of test code is a lot. Question, did this help you develop it?

Reply:
It's a lot because I copied them from a different class mostly because I didn't know what I was doing.

The issue here is there are no tests actually checking if there are checks on the handshakes.  All of the tests are whether the handshake works at a wire level, but there are none at the transport level.

I think a negative test is important to know that we are actually validating our connections.  The smoke test can handle the positive test.